webpack/scenes/Subscriptions/Manifest/CdnConfigurationTab/CustomCdnTypeForm.js

```diff
@@ -0,0 +1,148 @@
+import React, { useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { useDispatch, useSelector } from 'react-redux';
+import { FormattedMessage } from 'react-intl';
+import {
+  ActionGroup,
+  Button,
+  Form,
+  FormGroup,
+  FormSelect,
+  FormSelectOption,
+  TextInput,
+  Text,
+} from '@patternfly/react-core';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { noop } from 'foremanReact/common/helpers';
+
+import { CUSTOM_CDN } from './CdnConfigurationConstants';
+import { updateCdnConfiguration } from '../../../Organizations/OrganizationActions';
+import {
+  selectUpdatingCdnConfiguration,
+} from '../../../Organizations/OrganizationSelectors';
+import './CdnConfigurationForm.scss';
+
+const CustomCdnTypeForm = ({
+  showUpdate, onUpdate, contentCredentials, cdnConfiguration,
+}) => {
+  const dispatch = useDispatch();
+  const urlValue = cdnConfiguration.type === CUSTOM_CDN ? cdnConfiguration.url : '';
+  const [url, setUrl] = useState(urlValue);
+  const [updateEnabled, setUpdateEnabled] = useState(showUpdate);
+  const updatingCdnConfiguration = useSelector(state => selectUpdatingCdnConfiguration(state));
+  const firstUpdate = useRef(true);
+  const [sslCaCredentialId, setSslCaCredentialId] = useState(cdnConfiguration.ssl_ca_credential_id);
+
+  useEffect(() => {
+    if (firstUpdate.current) {
+      firstUpdate.current = false;
+      return;
+    }
+    setUpdateEnabled(true);
+  }, [sslCaCredentialId, cdnConfiguration]);
```

Comment:
Can you help me understand why this effect was needed?

Reply:
Enable/Disable update if `sslCaCredentialId` or `cdnConfiguration` changes? Are you questioning the `firstUpdate.current` part ?

Reply:
Right, why did it need an effect and why keep track of if it's the first update?

Is there a way that `updateEnabled` could be turned into a computed value rather than a state? If so it could make things much cleaner.

Reply:
https://beta.reactjs.org/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state